app/FileMapping.php

```diff
@@ -20,14 +22,18 @@ public function __construct()
 
         //get headers from the first line
         $headers = [];
-        $headers = fgetcsv($file, 0, chr(9));
+        $headers = fgetcsv($file, 0, $this->separator);
         $this->rows = 0;
-        while ($line = fgetcsv($file, 0, chr(9))) {
+        while ($line = fgetcsv($file, 0, $this->separator)) {
             $temp_array = [];
             for ($i = 0; $i < count($line); $i++) {
                 $key = $headers[$i];
                 $value = $line[$i];
-                $temp_array[$key] = $value;
+                if (isset($value) && $value != '') {
```

Comment:
Just a suggestion:
```
 $temp_array[$key] = null;
 if (isset($value) && $value != '') {
    $temp_array[$key] = $value;
}
```
It makes it more clear there's a default value, and if <something>, we set it to something else.